File: src/main/java/de/poeschl/bukkit/logcleaner/utils/InstanceFactory.java
package de.poeschl.bukkit.logcleaner.utils;

import de.poeschl.bukkit.logcleaner.helper.FileHelper;
import de.poeschl.bukkit.logcleaner.managers.SettingManager;
import de.poeschl.bukkit.logcleaner.threads.LogCleanerThread;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.plugin.java.JavaPlugin;

import java.io.File;
import java.util.logging.Logger;

/**
 * Project: logcleaner
 * Created by Markus on 26.06.2016.
 */
public class InstanceFactory {
    public SettingManager createSettingsManager(FileConfiguration config, Logger logger) {
        return new SettingManager(config, logger);
    }

    public Logger getLogger(JavaPlugin javaPlugin) {
        return javaPlugin.getLogger();
    }

    public FileHelper createFileHelper(Logger logger) {
        return new FileHelper(logger);
    }

    public LogCleanerThread createLogCleanerThread(FileHelper fileHelper, SettingManager settingManager, File logFolder) {
        return new LogCleanerThread(fileHelper, settingManager.getKeepDays(), logFolder);
    }
}

File: src/main/kotlin/de/poeschl/bukkit/logcleaner/utils/InstanceFactory.kt
package de.poeschl.bukkit.logcleaner.utils

import de.poeschl.bukkit.logcleaner.helper.FileHelper
import de.poeschl.bukkit.logcleaner.helper.FileHelperImpl
import de.poeschl.bukkit.logcleaner.managers.SettingsManager
import de.poeschl.bukkit.logcleaner.managers.SettingsManagerImpl
import de.poeschl.bukkit.logcleaner.threads.LogCleanerRunnable
import de.poeschl.bukkit.logcleaner.threads.LogCleanerRunnableImpl
import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.plugin.java.JavaPlugin
import java.io.File
import java.util.logging.Logger

open class InstanceFactory {
    open internal fun createSettingsManager(config: FileConfiguration, logger: Logger): SettingsManager {
        return SettingsManagerImpl(config, logger)
    }

    open internal fun getLogger(javaPlugin: JavaPlugin): Logger {
        return javaPlugin.logger
    }

    open internal fun createFileHelper(logger: Logger): FileHelper {
        return FileHelperImpl(logger)
    }

    open internal fun createLogCleanerRunnable(fileHelper: FileHelper, settingManager: SettingsManager, logFolder: File): LogCleanerRunnable {
        return LogCleanerRunnableImpl(fileHelper, settingManager.keepDays, logFolder)
    }
}
